content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions.md

```diff
@@ -693,6 +699,216 @@ kubectl apply -f my-crontab.yaml
+| `size(self.names) == size(self.details) && self.names.all(n, n in self.details)` | Validate the 'details' map is keyed by the items in the 'names' listSet           |
+
+Xref: [Supported evaluation on CEL](https://github.com/google/cel-spec/blob/v0.6.0/doc/langdef.md#evaluation)
+
+
+- If the Rule is scoped to the root of a resource, it may make field selection into any fields
+  declared in the OpenAPIv3 schema of the CRD as well as `apiVersion`, `kind`, `metadata.name` and
+  `metadata.generateName`. This includes selection of fields in both the `spec` and `status` in the
+  same expression, e.g. `self.status.quantity <= self.spec.maxQuantity`. 
+
+- If the Rule is scoped to an object with properties, the accessible properties of the object are field selectable
+  via `self.field` and field presence can be checked via `has(self.field)`. Null valued fields are treated as
+  absent fields in CEL expressions.
+
+- If the Rule is scoped to an object with additionalProperties (i.e. a map) the value of the map
+  are accessible via `self[mapKey]`, map containment can be checked via `mapKey in self` and all entries of the map
+  are accessible via CEL macros and functions such as `self.all(...)`.
+
+- If the Rule is scoped to an array, the elements of the array are accessible via `self[i]` and also by macros and
+  functions.
+
+- If the Rule is scoped to a scalar, `self` is bound to the scalar value.
+  
+Examples:
+
+|type of the field rule scoped to    | Rule example             |
+| -----------------------| -----------------------|
+| root object            | "self.status.actual <= self.spec.maxDesired"|
+| map of objects         | "self.components['Widget'].priority < 10"|
+| list of integers       | "self.values.all(value, value >= 0 && value < 100)"|
+| string                 | "self.startsWith('kube')"|
+
+
+The `apiVersion`, `kind`, `metadata.name` and `metadata.generateName` are always accessible from the root of the 
+object and from any x-kubernetes-embedded-resource annotated objects. No other metadata properties are accessible.
+	
+Unknown data preserved in custom resources via `x-kubernetes-preserve-unknown-fields` is not accessible in CEL
+  expressions. This includes:
+  - Unknown field values that are preserved by object schemas with x-kubernetes-preserve-unknown-fields.
+  - Object properties where the property schema is of an "unknown type". An "unknown type" is recursively defined as:
+    - A schema with no type and x-kubernetes-preserve-unknown-fields set to true
+    - An array where the items schema is of an "unknown type"
+    - An object where the additionalProperties schema is of an "unknown type"
+
+
+Only property names of the form `[a-zA-Z_.-/][a-zA-Z0-9_.-/]*` are accessible.
+Accessible property names are escaped according to the following rules when accessed in the expression:
+
+| property name          | escaped to             |
```

Comment:
Naming the column headers here is tricky. We want to convey that it's not just a property name of `__`, it's the character sequence of `__` _in_ any property name. Maybe switch which column is on which side?

| escape sequence | property name equivalent |
| -----------------------| ----------------------- |
| `__underscores__`                   | `__`      |
| `__dot__`                    | `.`              |

?

Reply:
Also, for the RESERVED keywords, the rule is more strict.  It _does_ need to match the exact property name (e.g. `int` in the word `sprint` would not be escaped)

Reply:
Updated and added notes.